yalafi/packages/cleveref.py

```diff
@@ -0,0 +1,215 @@
+\{
+    ([^\}\{]+)      # group 3: the label of the reference
+\}
+/
+    (.*)            # group 4: the replacement string
+/g
+''', re.VERBOSE)
+re_ref_range = re.compile(r'''
+s/\\
+(                   # group 1: '\cref' or '\Cref'
+    \\crefrange|\\Crefrange
+)
+(?:\\)?             # non capturing group
+(\*?)               # group 2: '*' if present, '' else
+\{
+    ([^\}\{]+)      # group 3: the label of the reference
+\}
+/
+    (.*)            # group 4: the replacement string
+/g
+''', re.VERBOSE)
+re_command = re.compile(r'''
+s/\\
+(                   # group 1: command
+    \\
+    (               # group 2: capturing cC
+        \[cC\]
+    )?
+    [a-zA-Z@]+
+)
+(                   # group 3: possible arguments
+    (?:\{\.\*\})+
+)?
+\s*                 # possible whitespaces inbetween
+/
+    (.*)            # group 4: replacement
+/g
+''', re.VERBOSE)
+re_cC = re.compile(r'(\[cC\])')
+re_escaped_dot = re.compile(r'\\\.')
+re_escaped_backslash = re.compile(r'\\\\')
+re_escaped_star = re.compile(r'\\\*')
+
+
+def re_remove_escaped_symbols(string):
+    r"""
+    Replaces certain strings that are escaped in sed file:
+    '\\.' -> '.'
+    '\\\\' -> '\\'
+    '\*' -> ''
+    """
+    string = re_escaped_dot.sub('.', string)
+    string = re_escaped_star.sub('', string)
+    string = re_escaped_backslash.sub(r'\\', string)
+    return string
+
+
+macro_read_sed = '\\YYCleverefInput'
+
+msg_poorman_option = f'''To use cleveref with YaLafi, you need to use the 'poorman' option
+*** and {macro_read_sed} to load the .sed file.
+'''
+
+msg_sed_not_loaded = f'''To use cleveref with YaLafi, you should use {macro_read_sed} to load
+*** the sed file, e.g. '{macro_read_sed}{{main.sed}}' if your LaTeX
+*** document is called 'main.tex'.
+'''
+
+
+def init_module(parser, options, position):
+    parms = parser.parms
+    parms.newcommand_ignore.append(macro_read_sed)
+
+    macros_latex = r'''
+        \newcommand{\crefname}[3]{}
+        \newcommand{\Crefname}[3]{}
+    '''
+
+    macros_python = [
+
+        Macro(parms, macro_read_sed, args='A', repl=h_read_sed),
+        Macro(parms, '\\label', args='OA', repl=''),
+
+        # Define functions which warn the User, whenever cleveref
+        # is used without invoking \YYCleverefInput. These will be overwritten
+        # by invoking \LTRreadSed.
+        Macro(parms, '\\cref', args='*A', repl=h_cref_warning),
+        Macro(parms, '\\Cref', args='*A', repl=h_cref_warning),
+        Macro(parms, '\\crefrange', args='*AA', repl=h_cref_warning),
+        Macro(parms, '\\Crefrange', args='*AA', repl=h_cref_warning),
+
+    ]
+
+    environments = []
+
+    # Warn the user, whenever cleveref is used
+    # without the poorman option:
+    if not is_poorman_used(options):
+        someVariable = utils.latex_error(msg_poorman_option, position, parser.latex, parms)
+
+    return InitModule(macros_latex=macros_latex, macros_python=macros_python,
+                      environments=environments)

```

Comment:
Here, I would propose the following. This will include the `LATEXXXERROR` mark into the plain text.
```
    # Warn the user, whenever cleveref is used
    # without the poorman option:
    if not is_poorman_used(options):
        inject_tokens = utils.latex_error(msg_poorman_option, position,
                                            parser.latex, parms)
    else:
        inject_tokens = []

    return InitModule(macros_latex=macros_latex, macros_python=macros_python,
                      environments=environments, inject_tokens=inject_tokens)
```

Reply:
That is very nice. I did not know that it is possible.

Reply:
The mistake is clearly mine, I only very recently added the mechanism with `inject_tokens` to `amsmath.py` that is referenced as example in `README.md`.